indexclient/client.py

```diff
@@ -280,18 +280,26 @@ def create_alias(
         resp = self._put(url, headers=headers, data=data, auth=self.auth)
         return resp.json()
 
-    def get_latest_version(self, did, skip_null_versions=False):
+    def get_latest_version(self, did, skip_null_versions=False, hide_deleted_urls=False):
         """
         Args:
             did (str): document id of an existing entry whose latest version is requested
             skip_null_versions (bool): if True, exclude entries without a version
+            hide_deleted_urls (bool): if True, remove deleted url endpoints from document
         Returns:
             Document: latest version of the entry
         """
 
         params = {"has_version": "true" if skip_null_versions else "false"}
         doc = self._get("index", did, "latest", params=params).json()
 
+        if doc and hide_deleted_urls:
+            urls_to_remove = [url for url, metadata in doc.urls_metadata.items()
```

Comment:
Good catch, I think it will be safer to check for that. I'll include it in the if statement.

Reply:
I think moving this solution to the indexd server (similar to how the skip_null_version flag was implemented) would be more consistent and possibly more efficient. If the latest version is marked as deleted for some reason it might make sense to return something else (a previous latest) and I think this logic is only removing urls metadata and still returning deleted doc